### PR TITLE
Fire WP Actions when mappings are created, updated and deleted

### PR DESCRIPTION
**Pull request by joehoyle**
I added a somewhat unusual clone of the prevous object, so the hook to
update can read the previous state of the mapping.

### CODE REVIEW

**Comment by joehoyle** [comment]
@rmccue #reviewmerge

File: class-mapping.php [hunk 1]
```diff
@@ -31,6 +31,10 @@ protected function __construct( $data ) {
 		$this->data = $data;
 	}
 
+	public function __clone() {
+		return $this->data = clone( $this->data );
```
Comment (rmccue) [inline class-mapping.php]: This shouldn't have `return`, and needs inline docs

**Comment by rmccue** [comment]
The actions here need docs too. :)

**Comment by joehoyle** [comment]
@rmccue take two!

**Comment by rmccue** [comment]
:ok_woman:

**Pull request merged**